bitcoin/BRWalletManager.h

```diff
@@ -98,8 +98,41 @@ bwmAnnounceSubmit (BRWalletManager manager,
 /// Transaction Event
 ///
 typedef enum {
+    // NOTE: The transaction is NOT owned by the event handler.
+    /**
+     * For P2P and API, this event occurs once a transaction has been created for a
+     * wallet, but is not yet signed or added.
+     */
+    BITCOIN_TRANSACTION_CREATED,
+
+    /**
+     * For P2P and API, this event occurs once a transaction has been signed for a
+     * wallet, but is not yet added.
+     */
+    BITCOIN_TRANSACTION_SIGNED,
+
+    /**
+     * For P2P and API, this event occurs once a transaction has been added to a
+     * wallet (via BRWalletRegisterTransaction). Transactions are added when they have
+     * been relayed by the P2P network or synced via BlockchainDB.
+     */
     BITCOIN_TRANSACTION_ADDED,
+
+    /**
+     * For P2P, this event occurs once a transaction has been marked as CONFIRMED/UNCONFIRMED
+     * by the P2P network.
+     *
+     * For API, this event does not occur as transactions are implicitly CONFIRMED when synced.
+     */
     BITCOIN_TRANSACTION_UPDATED,
+
+    /**
+     * For P2P, this event occurs once a transaction has been deleted from a wallet
+     * (via BRWalletRemoveTransaction) as a result of an UNCONFIRMED transaction no longer
+     * being visible in the mempools of any of connected P2P peers.
+     *
+     * For API, this event does not occur as transactions are implicitly CONFIRMED when synced.
```

Comment:
In light of conversations with Sam, this statement and assumption will likely need to be revisited. For now, however, it is an accurate statement from the point of view of how the code behaves.